### DISCUSSION TITLE
Custom Vehicle (Truck + Trailer)

### DISCUSSION THREAD

**Discussion by DanielAtt2000** [carla-simulator/carla#5624] [General]
Hi,

I am currently working on a Carla Simulation which requires a truck with trailer vehicle. As far as I am aware, this vehicle is not available in Carla.

Can anyone give me an insight into what is required to build such a vehicle from scratch?
Is it even possible?
What physics is involved?
Is building Carla suitable or do I require building carla with Unreal using docker?

Thank for your help.

**Comment by abhijeetbehera97** [comment]
Hi Daniel, 

Did you find a truck trailer model?

Did you use docker?

Thanks.

**Comment by IndefiniteBen** [comment]
This is stale, but I found some options while searching (and also this post). These only work with UE4 based CARLA (will likely need extensive work to function in UE5 CARLA). 

* This one is textured and seems more developed: https://github.com/KomasaQi/Carla-Truck-Trailer
* This one is untextured and hasn't been updated in years: https://github.com/frankeng/CarlaSemiTruckTrailer
